packages/medusa-js/src/request.ts

```diff
@@ -199,6 +200,9 @@ class Client {
     options: RequestOptions = {},
     customHeaders: Record<string, any> = {}
   ): Promise<any> {
+   
+    customHeaders = { ...customHeaders, ...this.config.customHeaders }
```

Comment:
suggestion: I think that we should invert the sense here, the custom headers from the config first and if some custom headers are provided when calling it then it should take priority. wdyt?

Reply:
Good point. Think it makes sense to allow developers to overwrite custom headers on a per-request basis.

Reply:
The intention is to allow override, but the order needs to be what was initially proposed for an override to work.  The second element overrides the first.

Reply:
The idea with the `customHeaders` configuration is global headers - headers shared among all requests - right? Wouldn't it make sense to allow overrides on a per-request basis, which I think is what Adrien proposes?

Reply:
Yes, this is what I proposed, so the config headers are applied but overridden by the request custom headers. With the current approach, if someone wants to change the custom headers from the config they won't be able to as they are applied at last